ledger/tests/shred.rs

```diff
@@ -135,7 +135,17 @@ fn test_multi_fec_block_different_size_coding() {
             .chain(fec_coding_shreds.iter().step_by(2))
             .cloned()
             .collect();
-        let recovered_data = Shredder::try_recovery(all_shreds, &reed_solomon_cache).unwrap();
+        let recovered_data: Vec<Shred> = shred::recover(all_shreds, &reed_solomon_cache)
+            .unwrap()
+            .filter_map(|s| {
+                let s = s.unwrap();
+                if s.is_data() {
+                    Some(s)
+                } else {
+                    None
+                }
+            })
```

Comment:
nit: I typically dislike the single letter abbreviations and would prefer `shred`. Also, you might be able to avoid the if/else with:
```rust
let shred = shred.unwrap();
shred.is_data().then_some(shred)
```

Reply:
Oh this is a neat trick, thanks! Surprised clippy would not offer it...